src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -764,4 +765,18 @@ default int getDefaultTooltipHideFlags(@Nonnull ItemStack stack)
         return 0;
     }
 
+    /**
+     * Get the food properties for this item.
+     * Use this instead of the {@link Item#getFoodProperties()} method, for ItemStack sensitivity.
+     *
+     * @param stack The ItemStack the entity wants to eat.
+     * @param entity The entity which wants to eat the food. Be aware that this can be null!
+     * @return The current FoodProperties for the item.
+     */
+    @Nullable
```

Comment:
Are there any restrictions on when this should be nullable? After looking at the PR in more depth it seems there are a few places patched that if a modder returns null from here but marks their item as edible a null pointer will be thrown. It is probably worth specifying in the javadocs for this that it shouldn't return null if the item returns true for is edible. (At least assuming that is actually true)

Reply:
Yeah I wrote about exactly this in the description. With the Nullable annotation it is exactly like Vanilla, since this NPE already is there.

Reply:
That is why I believed what I was saying is correct that there are restrictions, so then that definitely should be mentioned here (in the javadocs) to make sure modders know they shouldn't just return null unless their item is marked as never being edible.

Reply:
For me the current question is, should I just remove the Nullable annotation? There is no need for it and the method isn't really Nullable anyways

Reply:
I feel like yes as the same as the method it wraps around, it just should have a clarification in the javadocs that it only should be null if isEdible is false.